cmd/tsconnect/src/index.ts

```diff
@@ -4,55 +4,26 @@
 
 import "./wasm_exec"
 import wasmUrl from "./main.wasm"
-import { notifyState, notifyNetMap, notifyBrowseToURL } from "./notifier"
 import { sessionStateStorage } from "./js-state-store"
+import { renderApp } from "./app"
 
-const go = new Go()
-WebAssembly.instantiateStreaming(
-  fetch(`./dist/${wasmUrl}`),
-  go.importObject
-).then((result) => {
+async function main() {
+  const app = await renderApp()
+  const go = new Go()
```

Comment:
This is cool, how are you telling JS about Go without `globalThis`?

Reply:
It's added by the Go/WASM type definitions: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/golang-wasm-exec/index.d.ts

We specify the types as dev dependencies in package.json (https://github.com/tailscale/tailscale/blob/b81d219b24dc49ca91967837fddef274c71be51d/cmd/tsconnect/package.json#L6), and TypeScript automatically picks them up.